src/plugins/intel_cpu/src/edge.h

```diff
@@ -82,6 +82,7 @@ class Edge {
     }
 
     std::string name() const;
+    const MemoryDesc& getDesc() const;
```

Comment:
This is actually not necessary anymore.
Reverted.

Reply:
This is more robust only from the scope change standpoint. But it doesn't help the user of the interface to use the object in a particular way as it gives more degrees of freedom, which results in more possibilities of making a mistake. 
It looks like this method has been moved to the public scope to be accessed from the auxiliary static functions, which are established to keep graph code more clear. Even though the downsides of changing the scope of this method outweigh the benefits, it's clear that it would be excessive to solve the problem of accessing the memory descriptor via special routines that would basically do the same thing as this method. So we  have two solutions here
1. Convert the static routines to private Graph methods, so that `getDesc` call being private still remains accessible as Graph is still a "friend" of Edge class
2. Rename the method, so that it becomes clear for the user of the interface when to use it. "getDesc" -> "getCompileTimeDesc"/"getInitializationDesc" or something like that, just to help people to use the method properly and avoid mistakes.

Reply:
For me these all just two other workarounds, neither better nor worse.
The only 'real' fix for this problem is to make selectedPrimitiveDescriptor related data structures temporary (which they actually are). 
Collecting those descriptors, processing and applying them should be just an intermediate stage in a graph pipeline, like many other stages.
So, it should not be possible to access them simply because you don't have them at the inference stage at all.
And they should not be named "getCompileTimeDesc", because one will be able to access them only during CompileModel stage.
And still, currently nothing prevents the developer to access all these data structures using other public methods.
Nothing prevents Graph to access them at a wrong stage of the pipeline.

Reply:
Then let's help the interface users avoid unnecessary mistakes and give the method more specific name like "getOriginalDesc" or "getInitialDesc", or any other variate that will help to understand the purpose of this call.